packages/client/utils/parseEmailAddressList.ts

```diff
@@ -11,7 +19,38 @@ const parseEmailAddressList = (rawStr = '') => {
     .trim()
     // remove trailing commas
     .replace(/,$/g, '')
-  return emailAddresses.parseAddressList(commaDelimStr)
+
+  const emailCount = commaDelimStr.split(',').length
+  const lastChar = rawStr.charAt(rawStr.length - 1)
+  const isAddingNewEmail = lastChar !== ' ' && lastChar !== ','
+
+  if (emailCount === 1 && isAddingNewEmail) {
+    return {
+      parsedInvitees: emailAddresses.parseAddressList(commaDelimStr),
+      invalidEmailExists: false
+    }
+  }
+  const filteredStr = isAddingNewEmail ? filterLastEmail(commaDelimStr) : commaDelimStr
+  if (emailAddresses.parseAddressList(filteredStr)) {
+    return {
+      parsedInvitees: emailAddresses.parseAddressList(filteredStr),
+      invalidEmailExists: false
+    }
+  } else {
+    for (let i = filteredStr.length; i >= 0; i--) {
+      const slicedStr = filteredStr.slice(0, i)
+      if (emailAddresses.parseAddressList(slicedStr)) {
```

Comment:
should only parse once. work performed on keystroke will feel laggy on mobile, so we wanna keep it as cheap as possible